src/name_type.rs

```diff
@@ -92,11 +100,53 @@ impl fmt::Debug for NameType {
 }
 
 impl PartialEq for NameType {
+
     fn eq(&self, other: &NameType) -> bool {
         slice_equal(&self.0, &other.0)
     }
+
+}
+
+//FIXME(ben): the ID can be ordered from zero as a normal euclidean number
+//
+impl Ord for NameType {
+  #[inline]
+  fn cmp(&self, other : &NameType) -> Ordering {
+    Ord::cmp(&&self.0[..], &&other.0[..])    
+  }
+}
+
+impl PartialOrd for NameType {
+  #[inline]
+  fn partial_cmp(&self, other : &NameType) -> Option<Ordering> {
+    PartialOrd::partial_cmp(&&self.0[..], &&other.0[..])
+  }
+  #[inline]
+  fn lt(&self, other : &NameType) -> bool {
+    PartialOrd::lt(&&self.0[..], &&other.0[..])
+  }
+  #[inline]
+  fn le(&self, other : &NameType) -> bool {
+    PartialOrd::le(&&self.0[..], &&other.0[..])
+  }
+  #[inline]
+  fn gt(&self, other : &NameType) -> bool {
+    PartialOrd::gt(&&self.0[..], &&other.0[..])
+  }
+  #[inline]
+  fn ge(&self, other : &NameType) -> bool {
+    PartialOrd::ge(&&self.0[..], &&other.0[..])
+  }
 }
 
+// FIXME(Ben): please fix me
+impl hash::Hash for NameType {
+  fn hash<H: hash::Hasher>(&self, state: &mut H) {
+    state.write(&self.0[..])
```

Comment:
Am not sure either. But I think, Hash trait is not implemented for [T;64], from the documentation (https://doc.rust-lang.org/std/hash/trait.Hash.html). So not sure, whether hash() can be called directly as in the example.